elasticsearch/config.go

```diff
@@ -64,14 +63,6 @@ func DefaultConfig() *Config {
 // Hosts is an array of host strings and needs to have at least one entry
 type Hosts []string
 
-// Validate ensures Hosts instance has at least one entry
-func (h Hosts) Validate() error {
-	if len(h) == 0 {
-		return errInvalidHosts
-	}
-	return nil
-}
-
```

Comment:
Can you elaborate in which scenarios this `Validate` method is now called where `len(h)==0` would be ok?

Reply:
I don't know (hence the "hopefully" above). This definitely fixes this error:
`./apm-server apikey create -E apm-server.api_key.elasticsearch.username=admin -E apm-server.api_key.elasticsearch.password=changeme`
```Error processing configuration: `Hosts` must at least contain one hostname accessing 'apm-server.api_key.elasticsearch.hosts' (source:'command line flag')
```

If you think it is not related, I can create a separate ticket, link it to it, and leave #3298 open.